### PR TITLE
Add KMS Sign/Verify

### PR DESCRIPTION
**Pull request by josecoelho**
This is adding support for KMS [sign](https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html) and [verify](https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html).

I added support for `KeySpec`  keeping support for the deprecated deprecated `CustomerMasterKeySpec`. I can move that change to another PR if it makes things easier.

This is not the full implementation for sign and verify, as the algorithm and private key type are hardcoded. But it's enough to use `sign` and `verify` within the application. Would that be enough for now? It is good enough for our use case.

`GrantTokens` and `MessageType` `DIGEST` are not supported as well.

### CODE REVIEW

File: tests/test_kms/test_kms_boto3.py [hunk 1]
```diff
@@ -1106,3 +1117,101 @@ def test_key_tag_added_arn_based_happy():
...
+def test_sign_invalid_signing_algorithm():
+    client = boto3.client("kms", region_name="us-west-2")
+
+    key = client.create_key(Description="sign-key", KeyUsage="SIGN_VERIFY")
+    key_id = key["KeyMetadata"]["KeyId"]
+
+    message = "My message"
+    signing_algorithm = "INVALID"
+
+    try:
```
Comment (bblommers) [inline tests/test_kms/test_kms_boto3.py:1151]: Can we use the `pytest.raises` here instead - that makes it easy to verify that the correct error message is raised, not just that it fails.

See http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html#negative-tests

File: tests/test_kms/test_kms_boto3.py [hunk 2]
```diff
@@ -1106,3 +1117,101 @@ def test_key_tag_added_arn_based_happy():
...
+def test_sign_invalid_key_usage():
+    client = boto3.client("kms", region_name="us-west-2")
+
+    key = client.create_key(Description="sign-key", KeyUsage="ENCRYPT_DECRYPT")
+    key_id = key["KeyMetadata"]["KeyId"]
+
+    message = "My message"
+    signing_algorithm = "RSASSA_PSS_SHA_256"
+
+    try:
```
Comment (bblommers) [inline tests/test_kms/test_kms_boto3.py:1168]: Same

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @josecoelho, thanks for the PR! Any implementation is welcome, it doesn't have to support everything at the same time.

Please see my comments - it looks like some other tests are failing as well due to the removal of `CustomerMasterKeySpec`.

File: moto/kms/models.py [hunk 3]
```diff
@@ -516,5 +520,94 @@ def retire_grant(self, key_id, grant_id, grant_token) -> None:
...
+
+        if message_type == "DIGEST":
+            raise NotImplementedError(
+                "MessageType DIGEST not supported by kms_verify mock yet"
+            )
+
+    def sign(self, key_id, message, message_type, grant_tokens, signing_algorithm):
+        key = self.describe_key(key_id)
+
+        self.__ensure_implemented_sign_and_verify_args(grant_tokens, message_type)
```
[the remark this answers is not in the document]
Comment (josecoelho) [reply moto/kms/models.py]: I moved to warnings [here](https://github.com/spulec/moto/pull/5243/files#diff-8bf627e2436b027528752b0b1818a087eb16cf58b490f3e9adb26a2de6609aa8R670)

File: moto/kms/utils.py [hunk 4]
```diff
@@ -58,6 +59,18 @@ def generate_master_key():
     return generate_data_key(MASTER_KEY_LEN)
 
 
+def generate_private_key():
+    """Generate a private key to be used on asymmetric sign/verify.
+
+    NOTE: KeySpec is not taken into consideration and the key is always RSA_2048
```
[the remark this answers is not in the document]
Comment (josecoelho) [reply moto/kms/utils.py:65]: Thanks for the advice, I added more info to [`sign`](https://github.com/spulec/moto/pull/5243/files#diff-0514df96f93d76619859abfba375a4eb8ae5113221b9feb9c57ddc648255c59eR545) and [`verify`](https://github.com/spulec/moto/pull/5243/files#diff-0514df96f93d76619859abfba375a4eb8ae5113221b9feb9c57ddc648255c59eR572)

File: tests/test_kms/test_kms_boto3.py (hunk 1 above)
Comment (josecoelho) [reply tests/test_kms/test_kms_boto3.py]: Fixed

File: tests/test_kms/test_kms_boto3.py (hunk 2 above)
Comment (josecoelho) [reply tests/test_kms/test_kms_boto3.py]: Fixed

File: tests/test_kms/test_kms_boto3.py [hunk 5]
```diff
@@ -1106,3 +1117,101 @@ def test_key_tag_added_arn_based_happy():
...
+def test_sign_invalid_message():
+    client = boto3.client("kms", region_name="us-west-2")
+
+    key = client.create_key(Description="sign-key", KeyUsage="SIGN_VERIFY")
+    key_id = key["KeyMetadata"]["KeyId"]
+
+    message = ""
+    signing_algorithm = "RSASSA_PSS_SHA_256"
+
+    try:
```
[the remark this answers is not in the document]
Comment (josecoelho) [reply tests/test_kms/test_kms_boto3.py]: Fixed

**Comment by josecoelho** [comment]
@bblommers I still have some issues with [one of the tests](https://github.com/josecoelho/moto/blame/feature/kms-sign-verify/tests/test_kms/test_kms_boto3.py#L915).

It is raising `ExpiredTokenException` but I'm having a hard time to identify the reason. Would it be possible to help me out?
```python
botocore.exceptions.ClientError: An error occurred (ExpiredTokenException) when calling the CreateKey operation: The security token included in the request is expired
```

**Comment by josecoelho** [comment]
@bblommers fixed the failing test. I missed `@mock_kms` before [a test](https://github.com/josecoelho/moto/blob/ca862a2e31d0e7d56e90e59e9caf4f3cba17a28b/tests/test_kms/test_kms_boto3.py#L122).

File: moto/kms/models.py [hunk 6]
```diff
@@ -516,5 +520,93 @@ def retire_grant(self, key_id, grant_id, grant_token) -> None:
             key = self.describe_key(key_id)
             key.retire_grant(grant_id)
 
+    def __ensure_valid_sing_and_verify_key(self, key: Key):
```
Comment (lxop) [inline moto/kms/models.py:523]:
```suggestion
    def __ensure_valid_sign_and_verify_key(self, key: Key):
```

**Review by bblommers — APPROVED** [review]
LGTM - thanks again @josecoelho!

**Pull request merged**